src/main/scala/com/nuzigor/spark/sql/h3/GetBaseCellNumber.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Igor Nuzhnov
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.nuzigor.spark.sql.h3
+
+import com.nuzigor.h3.H3
+import org.apache.spark.sql.catalyst.expressions.codegen.{CodeGenerator, CodegenContext, ExprCode}
+import org.apache.spark.sql.catalyst.expressions.{Expression, ExpressionDescription, ImplicitCastInputTypes, NullIntolerant, UnaryExpression}
+import org.apache.spark.sql.types.{DataType, IntegerType, LongType}
+
+/**
+ * @param child h3 origin.
+ */
+@ExpressionDescription(
+  usage = "_FUNC_(h3) - Returns the base cell number for h3 index.",
+  arguments = """
+       Arguments:
+         * h3 -  h3 index
+             622485130170302463l
+     """,
+  examples = """
+       Examples:
+         > SELECT _FUNC_(622485130170302463l);
+         //TODO: Add output examples
+     """,
+  since = "0.7.0"
+)
+case class GetBaseCellNumber(child: Expression) extends UnaryExpression with ImplicitCastInputTypes with NullIntolerant {
+  override def inputTypes: Seq[DataType] = Seq(LongType)
+  override def dataType: DataType = IntegerType
+  override def nullable: Boolean = true
+
+  override protected def nullSafeEval(originAny: Any): Any = {
+    val origin = originAny.asInstanceOf[Long]
+    val baseCell = H3.getInstance().getBaseCellNumber(origin)
+    if (0 <= baseCell && baseCell <= 121) {
+      baseCell
+    } else {
+      null
+    }
+  }
+
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
```

Comment:
Another alternative is to implement codeGen which access the H3 object and use the builtin `getBaseCellNumber` but that is consistently 10% slower than below version (haven't investigated the actual difference). I can create a separate PR to introduce benchmark subproject if we want more of it

Reply:
Fixed. Test failed due to collection on wrong column